### PR TITLE
#159987699 Users should be able to see other users' profile and update their own profile

### PR DESCRIPTION
**Pull request by emekafredy**
# What does this PR do?
Create endpoints for users profile

# Description of Task to be completed?
### Modify the following endpoints
- User signup ``` POST api/users ```
- User login ``` POST api/users/login ```
### Create the following endpoints
- Get users profiles ``` GET api/users ```
- Get user profile by username ``` GET api/users/<username> ```
- Update user profile ``` PUT api/users/<userId> ```
### Write tests for users profile

# How should this be manually tested?
- Clone the repo
- cd into the 'Thor-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'
- Open postman to test endpoints

# Any background context you want to provide?
N/A
# What are the relevant pivotal tracker stories?
[159987699](https://www.pivotaltracker.com/story/show/159987699)

# Screenshots (if appropriate)
N/A

# Questions:
N/A

### CODE REVIEW

File: server/tests/controllers/users_profile.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+};
+
+describe('Test for users profiles', () => {
+  describe('GET /api/users', () => {
+    it('should return all users profile', (done) => {
+      chai.request(app)
+        .get('/api/users')
+        .end((error, res) => {
+          expect(res).to.have.status(200);
+          res.body.profiles[0].should.have.property('firstName');
```
Comment (CEOehis) [inline server/tests/controllers/users_profile.js]: @emekafredy you can reduce code repetition by destructuring `profiles` from the `res.body`. Same thing for lines 46-48.

File: server/tests/controllers/users_profile.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          done();
+        });
+    });
+    it('should return a bad request error message for name input longer than required', (done) => {
+      chai.request(app)
+        .put('/api/users/1')
+        .send(incorrectNameInput)
+        .end((error, res) => {
+          expect(res).to.have.status(400);
+          res.body.errors.firstName.should.eql('First name should not be more than 20 characters');
```
Comment (CEOehis) [inline server/tests/controllers/users_profile.js]: @emekafredy I think you can also use destructuring here to extract `firstName` and `lastName` from `res.body.errors` for cleaner syntax.

File: index.js [hunk 3]
```diff
@@ -39,17 +42,16 @@ app.use(
 if (!isProduction) {
   app.use(errorhandler());
 }
+app.use(router);
 
-app.use(require('./server/routes'));
-
-// catch 404 and forward to error handler
+// / catch 404 and forward to error handler
```
Comment (CEOehis) [inline index.js]: @emekafredy do check the extra `/` in this comment. Same with line 54.

**Review by CEOehis — CHANGES_REQUESTED** [review]

File: server/tests/controllers/users_profile.js (hunk 1 above)
Comment (emekafredy) [reply server/tests/controllers/users_profile.js]: Thanks man. Will work on these

File: index.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+import bodyParser from 'body-parser';
+import session from 'express-session';
+import cors from 'cors';
+import errorhandler from 'errorhandler';
+import log from 'fancy-log';
+
+import router from './server/routes';
+
+const swaggerUi = require('swagger-ui-express'),
   swaggerDocument = require('./server/docs/swagger.json');
```
Comment (lauragift21) [inline index.js]: You can also use the import statement for swagger-ui-express package for consistency.

Comment (emekafredy) [reply index.js]: Thanks for calling my attention on this

**Review by CEOehis — APPROVED** [review]

**Pull request merged**